[pt2e][quant] Update some docs for pt2 export quantization (#2530)
pytorch/tutorials: prototype_source/pt2e_quantizer.rst

``SharedQuantizationSpec`` is designed for this use case to annotate tensors whose quantization
parameters are shared with other tensors. Input of ``SharedQuantizationSpec`` is an ``EdgeOrNode`` object which 
can be an input edge or an output value.

.. note::

>    * Sharing is Transitive

Reviewer: ```suggestion
   * Sharing is transitive
```

>      Some Tensors might be effectively be using shared quantization spec due to (1) two nodes/edges are

Reviewer: ```suggestion

     Some Tensors might be effectively using shared quantization spec due to (1) two nodes/edges are
```

     configured to use SharedQuantizationSpec (2) there is existing sharing of some of the nodes

>      For example, let's say we have two conv nodes conv1 and conv2, and both of them are fed into a cat

Reviewer: ```suggestion
     For example, let's say we have two ``conv`` nodes ``conv1`` and ``conv2``, and both of them are fed into a ``cat``
```

>      node. `cat([conv1_out, conv2_out], ...)` Let's say output of conv1, conv2 and first input of cat are configured

Reviewer: ```suggestion
     node. `cat([conv1_out, conv2_out], ...)` Let's say output of ``conv1``, ``conv2`` and the first input of ``cat`` are configured
```

>      with the same configurations of QuantizationSpec, second input of cat is configured to use SharedQuantizationSpec

Reviewer: ```suggestion
     with the same configurations of ``QuantizationSpec``, second input of ``cat`` is configured to use ``SharedQuantizationSpec``
```

>      with the first input.
>      conv1_out: qspec1(dtype=torch.int8, ...)
>      conv2_out: qspec1(dtype=torch.int8, ...)
>      cat_input0: qspec1(dtype=torch.int8, ...)
>      cat_input1: SharedQuantizationSpec((conv1, cat))  # conv1 node is the first input of cat

Reviewer: ```suggestion
     with the first input.
     
     .. code-block::
     
       conv1_out: qspec1(dtype=torch.int8, ...)
       conv2_out: qspec1(dtype=torch.int8, ...)
       cat_input0: qspec1(dtype=torch.int8, ...)
       cat_input1: SharedQuantizationSpec((conv1, cat))  # conv1 node is the first input of cat
```

>      First of all, the output of conv1 are implicitly sharing quantization parameter (and observer object)

Reviewer: ```suggestion
     First of all, the output of ``conv1`` is implicitly sharing quantization parameter (and observer object)
```

>      with first input of cat, and same for output of conv2 and second input of cat.

Reviewer: ```suggestion
     with the first input of ``cat``, and same for output of ``conv2`` and the second input of ``cat``.
```

>      So since user configures the two input of cat to share quantization parameters, by transitivity,

Reviewer: ```suggestion
     So since user configures the two inputs of ``cat`` to share quantization parameters, by transitivity,
```

>      conv2_out and conv1_out will also be sharing quantization parameters. In the observed graph, you

Reviewer: ```suggestion
     ``conv2_out`` and ``conv1_out`` will also be sharing quantization parameters. In the observed graph, you
```

     will see:

>      ```

Reviewer: ```suggestion
     .. code-block::
     
```

     conv1 -> obs -> cat
     conv2 -> obs   /

>      ```

Reviewer: ```suggestion

```

>      and both `obs` will be the same observer instance

Reviewer: ```suggestion

     and both ``obs`` will be the same observer instance.
```

``SharedQuantizationSpec`` is designed for this use case to annotate tensors whose quantization
parameters are shared with other tensors. Input of ``SharedQuantizationSpec`` is an ``EdgeOrNode`` object which 
can be an input edge or an output value.

> .. note::

Reviewer: ```suggestion
.. note::

```

   * Sharing is transitive

>      Some Tensors might be effectively using shared quantization spec due to (1) two nodes/edges are
>      configured to use SharedQuantizationSpec (2) there is existing sharing of some of the nodes
> 

Reviewer: ```suggestion
     Some tensors might be effectively using shared quantization spec due to:
     
     * Two nodes/edges are configured to use ``SharedQuantizationSpec``.
     * There is existing sharing of some nodes.
     
```
The way it's written right now is confusing. I suggested an overwrite above.

     For example, let's say we have two ``conv`` nodes ``conv1`` and ``conv2``, and both of them are fed into a ``cat``

>      node. `cat([conv1_out, conv2_out], ...)` Let's say output of ``conv1``, ``conv2`` and the first input of ``cat`` are configured

Reviewer: ```suggestion
     node: ``cat([conv1_out, conv2_out], ...)``. Let's say the output of ``conv1``, ``conv2``, and the first input of ``cat`` are configured
```

>      with the same configurations of ``QuantizationSpec``, second input of ``cat`` is configured to use ``SharedQuantizationSpec``

Reviewer: ```suggestion
     with the same configurations of ``QuantizationSpec``. The second input of ``cat`` is configured to use ``SharedQuantizationSpec``
```

     with the first input.
     
     .. code-block::
     
       conv1_out: qspec1(dtype=torch.int8, ...)
       conv2_out: qspec1(dtype=torch.int8, ...)
       cat_input0: qspec1(dtype=torch.int8, ...)
       cat_input1: SharedQuantizationSpec((conv1, cat))  # conv1 node is the first input of cat
     

>      First of all, the output of ``conv1`` is implicitly sharing quantization parameter (and observer object)
>      with the first input of ``cat``, and same for output of ``conv2`` and the second input of ``cat``.

Reviewer: ```suggestion
     First of all, the output of ``conv1`` is implicitly sharing quantization parameters (and observer object)
     with the first input of ``cat``, and the same is true for the output of ``conv2`` and the second input of ``cat``.
```

>      So since user configures the two inputs of ``cat`` to share quantization parameters, by transitivity,
>      ``conv2_out`` and ``conv1_out`` will also be sharing quantization parameters. In the observed graph, you
>      will see:

Reviewer: ```suggestion
     Therefore, since the user configures the two inputs of ``cat`` to share quantization parameters, by transitivity,
     ``conv2_out`` and ``conv1_out`` will also be sharing quantization parameters. In the observed graph, you
     will see the following:
     
```

     .. code-block::
     

>      conv1 -> obs -> cat
>      conv2 -> obs   /

Reviewer: ```suggestion
         conv1 -> obs -> cat
         conv2 -> obs   /
```

Reviewer: The output looks broken, adding correct indentation and extra lines should help:
<img width="870" alt="Screenshot 2023-08-23 at 3 34 15 PM" src="https://github.com/pytorch/tutorials/assets/5317992/14a714de-3320-4734-9669-7c56fea85bac">
